overlord/snapstate/handlers.go

```diff
@@ -177,8 +197,21 @@ func (m *SnapManager) doPrerequisites(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// check that we are not already queued it for install
+	for _, tc := range t.Change().Tasks() {
+		if tc.Kind() == "link-snap" {
+			snapsup, err := TaskSnapSetup(tc)
+			if err != nil {
+				return err
+			}
+			if snapsup.Name() == prereqName {
+				return nil
+			}
+		}
+	}
```

Comment:
I missed understanding this bit, so for link-snap we return nil, as we discussed then there's no waiting setup though which is wrong...  also I'm not sure this bit is fully right if there's also a discard-snap in the change